src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -207,6 +208,7 @@
     Map<String, WebMapServer> wmsCache;
     Map<String, WebMapTileServer> wmtsCache;
     Map<CoverageHintReaderKey, GridCoverageReader> hintCoverageReaderCache;
+    Map<StyleInfo,StyledLayerDescriptor> sldCache;
```

Comment:
So this is the more generic version of styleCache, can we implement styleCache in terms of sldCache and avoid duplication?

Reply:
Since getting a "Style" involves changing the name directly in the style object, we probably want to avoid it, although I am honestly not sure.

Reply:
Come to think of it, I need to fix the current implementation so we don't do that.

If we were to delegate, we would need to do a bit of post-processing each time we get the style - get the first namedLayer, then change the name. No disk lookups, so that might be okay.
I'll change it for now to delegate, and we can test during the beta to see if anything weird happens.

Reply:
Ok, I tried out integrating them, and ran into some issues (bug in StyleImpl.clone() - normally i'd fix it, but with the code freeze this close I don't really want to be messing around in core GeoTools), so for now I am just going to fully separate styleCache and sldCache. styleCache is likely to be much more frequently used, so this shouldn't be much of an issue from a memory / efficiency perspective.

Reply:
Use duplicating style visitor, rather than clone()